File: mobile/lib/features/profile/infrastructure/demo_member_profile_gateway.dart
import 'package:cnpm_mobile/features/profile/domain/member_profile.dart';
import 'package:cnpm_mobile/features/profile/domain/member_profile_gateway.dart';

final class DemoMemberProfileGateway implements MemberProfileGateway {
  const DemoMemberProfileGateway();

  @override
  Future<MemberProfileResult> loadProfile() async {
    return MemberProfileAvailable(
      MemberProfileSnapshot(
        displayLabel: 'Membre de démonstration',
        roleLabel: 'Administrateur fictif de l’entreprise',
        organizationName: 'Entreprise Démo Sahel',
        memberReference: 'CNPM-DEMO-0001',
        organizationTypeLabel: 'Entreprise fictive',
        membershipLabel: 'Adhésion active — scénario',
        membershipSince: DateTime.utc(2024, 3, 18),
        disclosure:
            'Projection locale fictive et non modifiable. Aucune coordonnée personnelle ni donnée membre réelle n’est affichée.',
      ),
    );
  }
}

File: web/src/app/features/member/profile/demo-member-profile.gateway.ts
import { Injectable } from '@angular/core';
import { delay, of, type Observable } from 'rxjs';
import type { MemberProfileGateway, MemberProfileSnapshot } from './member-profile-gateway';

const DEMO_PROFILE: MemberProfileSnapshot = {
  displayLabel: 'Responsable adhésion',
  roleLabel: 'Administrateur de l’entreprise',
  organizationName: 'Sahel Agro SA',
  memberReference: 'CNPM-2026-0001',
  organizationTypeLabel: 'Société anonyme',
  membershipLabel: 'Adhésion active',
  membershipSince: '2024-03-18',
  disclosure: 'Profil consultable en lecture seule.',
};

@Injectable()
export class DemoMemberProfileGateway implements MemberProfileGateway {
  load(): Observable<MemberProfileSnapshot> {
    return of(DEMO_PROFILE).pipe(delay(0));
  }
}
